sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -112,6 +125,28 @@ private static DoFnSignature parseSignature(Class<? extends DoFn<?, ?>> fnClass)
         findAnnotatedMethod(errors, DoFn.GetRestrictionCoder.class, fnClass, false);
     Method newTrackerMethod = findAnnotatedMethod(errors, DoFn.NewTracker.class, fnClass, false);
 
+    Collection<Method> onTimerMethods =
+        declaredMethodsWithAnnotation(DoFn.OnTimer.class, fnClass, DoFn.class);
+    HashMap<String, DoFnSignature.OnTimerMethod> onTimerMethodMap =
+            Maps.newHashMapWithExpectedSize(onTimerMethods.size());
+    for (Method onTimerMethod : onTimerMethods) {
+      String id = onTimerMethod.getAnnotation(DoFn.OnTimer.class).value();
+        errors.checkArgument(timerDeclarations.containsKey(id),
+            "Callback %s is for for undeclared timer %s", onTimerMethod, id);
+      onTimerMethodMap.put(id, OnTimerMethod.create(onTimerMethod, id, Collections.EMPTY_LIST));
+    }
+    builder.setOnTimerMethods(onTimerMethodMap);
+
+    // Check the converse - that all timers have a callback. This could be relaxed to only
+    // those timers used in methods, once method parameter lists support timers.
```

Comment:
I talked about this with @bjchambers and @dhalperi and we came to a consensus that it is useful for reviewers to have some documentation - could just be Java comments - and mostly harmless at this stage in Beam to have some Javadoc that describes a prospective feature, as long as it has the needed caveats.

So rather than convert it to a plain comment, I added caveats.